api/payments.go

```
@@ -176,34 +223,86 @@ func (a *API) PaymentCreate(w http.ResponseWriter, r *http.Request) error {
 		return internalServerError("There was an error charging your card: %v", err).WithInternalError(err)
 	}
 
-	// mark order and transaction as paid
-	tr.Status = models.PaidState
-	tx.Create(tr)
-	order.PaymentProcessor = provider.Name()
-	order.PaymentState = models.PaidState
-	order.InvoiceNumber = invoiceNumber
-	tx.Save(order)
+	paymentComplete(r, tx, tr, order)
+	tx.Commit()
```

swallowing the error I think?